src/NodaTime/Calendars/BclWeekYearRule.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright 2016 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using JetBrains.Annotations;
+using NodaTime.Annotations;
+using NodaTime.Extensions;
+using NodaTime.Utility;
+using System;
+using System.Globalization;
+using static System.Globalization.CalendarWeekRule;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    /// Implementation of <see cref="IWeekYearRule"/> to emulate the behavior of
+    /// <see cref="Calendar.GetWeekOfYear(DateTime, CalendarWeekRule, DayOfWeek)"/>.
+    /// </summary>
+    [Immutable]
+    public sealed class BclWeekYearRule : IWeekYearRule
+    {
+        private readonly CalendarWeekRule rule;
+        private readonly IsoDayOfWeek firstDayOfWeek;
+        private readonly int minDaysInFirstWeek;
+
+        // TODO: Expose properties?
+
+        private BclWeekYearRule(CalendarWeekRule rule, DayOfWeek firstDayOfWeek)
+        {
+            this.rule = rule;
+            this.firstDayOfWeek = firstDayOfWeek.ToIsoDayOfWeek();
+            switch (rule)
+            {
+                case FirstDay:
+                    minDaysInFirstWeek = 1;
+                    break;
+                case FirstFourDayWeek:
+                    minDaysInFirstWeek = 4;
+                    break;
+                case FirstFullWeek:
+                    minDaysInFirstWeek = 7;
+                    break;
+                default: throw new ArgumentException($"No such CalendarWeekRule: {rule}", nameof(rule));
```

Comment:
We cover all the defined rules; this would be an unknown enum value like `(CalendarWeekRule) 100`.

Will change the formatting for the throw - at one point this was in a factory method and all the cases were returns.